### PR TITLE
feat:  Offers tab in `profile` and `gallery item`

### LINKED ISSUE #10845: 3. Offers tab in gallery item
linked with https://github.com/kodadot/nft-gallery/issues/10807

# Offers tab in gallery item
slight refresh for the gallery item offer tab
image

- it will be the second tab 
- there is amount, expiration, from and a button for accept if you are the owner of the nft 

## Mobile layout
- added in there
image

### LINKED ISSUE #10846: 4. cancel / accept offer
## Cancelling offer
image

Cancel offer is the same button as for unfollow, without hover effect, just the cursor pointer 

## Accept offer

image

more details cc @exezbcz

### LINKED ISSUE #10912: 5. Manage offers on profile page
LINKED TO https://github.com/kodadot/nft-gallery/issues/10807

image

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

test: https://deploy-preview-10911--koda-canary.netlify.app/ahp/gallery/166-2408213503

- [x] Closes #10845
- [x] Closes #10846
- [x] Closes #10912

## Needs Design check

- @exezbcz please review

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

## Profile detail
CleanShot 2024-08-31 at 17 57 18@2x

## Cancel Offer
CleanShot 2024-08-31 at 18 00 41@2x

## Accept Offer
CleanShot 2024-08-31 at 18 01 24@2x

### CODE REVIEW

File: components/gallery/GalleryItemTabsPanel/GalleryItemOffersTable.vue [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    const { offers: offersData, loading: offersLoading } = useOffers({
+      where: { id_in: newOffers.map(offer => offer.id) },
+    })
+
+    stopWatch.value = watchEffect(() => {
+      loading.value = offersLoading.value
+      offers.value = offersData.value
+    })
+  },
+})
```
[the remark this answers is not in the document]
Comment (hassnian) [reply components/gallery/GalleryItemTabsPanel/GalleryItemOffersTable.vue:155]: see https://github.com/kodadot/nft-gallery/pull/10925#discussion_r1740259183

File: components/profile/activityTab/Offers.vue [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+      totalCount
+    }
+  `,
+  onChange: ({ data }) => {
+    counts.value = {
+      incoming: data.incoming.totalCount,
+      outgoing: data.outgoing.totalCount,
+    }
+  },
+})
```
[the remark this answers is not in the document]
Comment (hassnian) [reply components/profile/activityTab/Offers.vue:163]: ts issue - eslint: Expected to return a value in computed function.

File: components/offer/OfferOverviewModal.vue [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+        state: computed(() => session.value.state),
+        action: computed(() => {
+          if (isSessionState('succeeded')) {
+            return {
+              label: t('offer.manageOffers'),
+              icon: 'arrow-up-right',
+              url: `/${urlPrefix.value}/u/${accountId.value}`,
+            }
+          }
+          return undefined
```
[the remark this answers is not in the document]
Comment (hassnian) [reply components/offer/OfferOverviewModal.vue:273]: ts issue - eslint: Expected to return a value in computed function.

**Review by Jarsen136 — COMMENTED** [review]
✅ Works for me: accepting offer \ canceling offers.

Some small issues I found:

1. Decimals in price should be displayed in full. It looks good on the gallery details page.
2. the format of expired time could be improved. It could be the same as in the nft activity table. wdyt? @exezbcz 
3. At the third row of offers, the time looks incorrect.   17.08 (16 days ago?)
image


---

- amount decimals example:
image



- time format example:

image

**Comment by hassnian** [comment]
> 1. Decimals in price should be displayed in full. It looks good on the gallery details page.

fixed

CleanShot 2024-09-03 at 15 31 30@2x


> 3. At the third row of offers, the time looks incorrect.   17.08 (16 days ago?)

seems like an indexer issue, query is ok, that offer should be `EXPIRED`  cc @vikiival 

`````
      {
        "id": "174-4008829562",
        "expiration": "6920253",
        "blockNumber": "6919953",
        "price": "1000000",
        "status": "ACTIVE",
        "caller": "16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z",
        "desired": {
          "id": "166-2408213503",
          "name": "Ezop #56",
          "sn": "2408213503",
          "currentOwner": "13zFVp9pdsJCJPeVBBRiSAkhBgZ4qENVEZBvfbs9Ft4rHVM8",
          "__typename": "NFTEntity"
        },
        "__typename": "Offer"
      }

`````

File: components/profile/activityTab/Offers.vue [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  filters.value = filters.value.map(filter => filter.id === 'outgoing' ? ({ ...filter, active: true }) : filter)
+}
+
+const where = computed(() => {
+  const conditions = [] as Record<string, unknown>[]
+
+  const outgoing = {
+    caller_eq: props.id,
+    status_eq: 'ACTIVE',
+  }
```
Comment (Jarsen136) [inline components/profile/activityTab/Offers.vue:125]: I would display all the offers including active and expired ones, so I can withdraw the deposit from the expired offer.

I guess the deposit in the offer will not automatically return when it becomes expires.

**Review by Jarsen136 — COMMENTED** [review]
>> At the third row of offers, the time looks incorrect. 17.08 (16 days ago?)
> seems like an indexer issue, query is ok, and that offer should be EXPIRED 

image
I mean the days look confusing and it can't tell whether it's the future or the past
Is it possible to show the relative time of the offer?
like (16 days ago) ,  (in 3 days)

File: components/profile/activityTab/Offers.vue (hunk 4 above)
Comment (hassnian) [reply components/profile/activityTab/Offers.vue]: done

**Comment by hassnian** [comment]
> I mean the days look confusing and it can't tell whether it's the future or the past Is it possible to show the relative time of the offer? like (16 days ago) , (in 3 days)

I did talk with @exezbcz  about this and he wanted it without the "in x days" just "x days", added `time ago` if the offer is expired

**Review by Jarsen136 — APPROVED** [review]

**Comment by vikiival** [comment]
Yup it is a bug 
- https://github.com/kodadot/stick/issues/300

**Review by preschian — APPROVED** [review]

**Comment by vikiival** [comment]
Screenshot 2024-09-09 at 09 08 14

Hey, seems like a lot of tests is failing on this PR

**Comment by hassnian** [comment]
> Hey, seems like a lot of tests is failing on this PR

fixed tests related to this pr

**Pull request merged**